From oliver@neukum.org  Fri Jan 15 10:05:34 2010
From: Oliver Neukum <oliver@neukum.org>
Date: Wed, 23 Dec 2009 19:22:19 +0100
Subject: USB: Document ABI for USB_QUIRK_RESET_MORPHS
To: Greg KH <greg@kroah.com>, Linux USB list <linux-usb@vger.kernel.org>
Message-ID: <200912231922.19894.oliver@neukum.org>


From: Oliver Neukum <oliver@neukum.org>

Adds documentation about the attribute that tells
the kernel a device cannot be successfully reset.

Signed-off-by: Oliver Neukum <oliver@neukum.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 Documentation/ABI/testing/sysfs-bus-usb |   11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/Documentation/ABI/testing/sysfs-bus-usb
+++ b/Documentation/ABI/testing/sysfs-bus-usb
@@ -159,3 +159,14 @@ Description:
 		device.  This is useful to ensure auto probing won't
 		match the driver to the device.  For example:
 		# echo "046d c315" > /sys/bus/usb/drivers/foo/remove_id
+
+What:		/sys/bus/usb/drivers/.../reset
+Date:		December 2009
+Contact:	Oliver Neukum <oliver@neukum.org>
+Description:
+		Writing 1 to this file tells the kernel that this
+		device will morph into another mode when it is reset.
+		Drivers will not use reset for error handling for
+		such devices.
+Users:
+		usb_modeswitch
